src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag

```diff
@@ -45,6 +52,34 @@
       </c:choose>:
   </div>
   <c:choose>
+    <c:when test="${questionWithResponses.feedbackParticipantCommentsOnResponsesAllowed}">
+      <button type="button" class="btn btn-default btn-xs icon-button pull-right show-frc-${addOrEdit}-form"
+              id="button_add_comment-${responseRecipientIndex}-${responseGiverIndex}-${questionIndex}"
+              data-recipientindex="${responseRecipientIndex}" data-giverindex="${responseGiverIndex}"
+              data-qnindex="${questionIndex}" data-toggle="tooltip" data-placement="top"
+              title="<%=Const.Tooltips.COMMENT_ADD%>"
+              <c:if test="${data.preview or (not data.submittable)}"> disabled style="background: #66727A;"</c:if>>
+        <span class="glyphicon glyphicon-comment glyphicon-primary"></span>
+      </button>
+      <div class="${divClassType}">
+          ${response.submissionFormHtml}
+        <br>
+        <ul class="list-group" id="responseCommentTable-${responseRecipientIndex}-${responseGiverIndex}-${questionIndex}"
+            style="${not empty response.feedbackParticipantCommentOnResponse ? 'margin-top:15px;': 'display:none'}">
+          <c:if test="${isCommentOnResponse}">
+            <shared:feedbackResponseCommentRow frc="${response.feedbackParticipantCommentOnResponse}"
+                firstIndex="${responseRecipientIndex}" secondIndex="${responseGiverIndex}"
+                thirdIndex="${questionIndex}" frcIndex="1" moderatedPersonEmail="${moderatedPersonEmail}"
+                isSessionOpenForSubmission="${isSessionOpenForSubmission}" isPreview="${data.preview}"
+                submittable="${data.submittable}" isModeration="${data.moderation}"/>
+          </c:if>
+
+          <shared:feedbackResponseCommentAdd frc="${response.feedbackResponseCommentAdd}" firstIndex="${responseRecipientIndex}"
```

Comment:
In the case that there is no response ID, can just use your hack. No need to add new attribute as all variable are available in response and questionWithResponses